integrations/openai/src/databricks_openai/agents/session.py

```diff
@@ -0,0 +1,329 @@
+
+    def _ensure_session(self) -> None:
+        """Ensure the session record exists in agent_sessions table."""
+        now = datetime.now(timezone.utc)
+
+        with self._pool.connection() as conn:
+            conn.execute(
+                f"""
+                INSERT INTO {self.sessions_table} (session_id, created_at, updated_at)
+                VALUES (%s, %s, %s)
+                ON CONFLICT (session_id) DO NOTHING
+                """,
+                (self.session_id, now, now),
+            )
+
+        logger.debug(f"Ensured session {self.session_id} exists")
+
+    async def get_items(self, limit: int | None = None) -> list[TResponseInputItem]:
+        """
+        Retrieve the conversation history for this session.
+
+        Args:
+            limit: Maximum number of items to retrieve. If None, retrieves all items.
+                   When specified, returns the latest N items in chronological order.
+
+        Returns:
+            List of input items representing the conversation history.
+        """
+        if limit is not None:
+            # Get the last N items, but return in chronological order
+            query = f"""
+                SELECT message_data FROM (
+                    SELECT message_data, id
+                    FROM {self.messages_table}
+                    WHERE session_id = %s
+                    ORDER BY id DESC
+                    LIMIT %s
+                ) sub
+                ORDER BY id ASC
+            """
+            params = (self.session_id, limit)
+        else:
+            # Get all items in chronological order
+            query = f"""
+                SELECT message_data 
+                FROM {self.messages_table}
+                WHERE session_id = %s
+                ORDER BY id ASC
+            """
+            params = (self.session_id,)
+
+        with self._pool.connection() as conn:
+            result = conn.execute(query, params)
+            rows = result.fetchall()
+
+        items = []
+        for row in rows:
+            message_data = row["message_data"]
+            # Parse JSON string if needed (psycopg may return JSONB as string)
+            if isinstance(message_data, str):
+                items.append(json.loads(message_data))
+            else:
+                items.append(message_data)
+
+        return items
+
+    async def add_items(self, items: list[TResponseInputItem]) -> None:
+        """
+        Add new items to the conversation history.
+
+        Args:
+            items: List of input items to add to the history.
+        """
+        if not items:
+            return
+
+        now = datetime.now(timezone.utc)
+
+        with self._pool.connection() as conn:
+            # Insert all messages
+            with conn.cursor() as cur:
+                cur.executemany(
+                    f"""
+                    INSERT INTO {self.messages_table} (session_id, message_data, created_at)
```

Comment:
`created_at` probably should be column default when creating the table. no need to set it explicitly.